buildSrc/src/main/java/com/r3/csde/VNodesHelper.java

```diff
@@ -0,0 +1,288 @@
+                            " \"corda.notary.service.plugin\" : \"net.corda.notary.NonValidatingNotary\" } }";
+        }
+
+        HttpResponse<JsonNode> response = Unirest.post(pc.baseURL + "/api/v1/membership/" + shortHash)
+                .body(registrationBody)
+                .basicAuth(pc.rpcUser, pc.rpcPasswd)
+                .asJson();
+
+        if (response.getStatus() == HTTP_OK) {
+            pc.out.println("Membership requested for node " + shortHash);
+        } else {
+            throw new CsdeException("Failed to register virtual node "+ shortHash +
+                    ", response status: " + response.getStatus() );
+            }
+
+        // wait until Vnode registered
+        pollForRegistration(shortHash, 30000, 1000);
+
+    }
+
+    /**
+     * Checks if a virtual node with given shortHash has been registered
+     * @param shortHash shortHash of the node which is being checked.
+     * @return returns true if the vnode is registered
+     */
+    private boolean checkVNodeIsRegistered(String shortHash) throws CsdeException {
+
+        // Queries registration status for vnode.
+        HttpResponse<JsonNode> response = Unirest.get(pc.baseURL + "/api/v1/membership/" + shortHash)
+                .basicAuth(pc.rpcUser, pc.rpcPasswd)
+                .asJson();
+
+        if(response.getStatus() != HTTP_OK)
+            throw new CsdeException("Failed to check registration status for virtual node '" + shortHash +
+                    "' response status: " + response.getStatus());
+
+        try {
+            // If the response body is not empty check all previous requests for an "APPROVED"
+            if (!response.getBody().getArray().isEmpty()) {
+                List<RegistrationRequestProgressDTO> requests = mapper.readValue(
+                        response.getBody().toString(), new TypeReference<>() {
+                        });
+                for (RegistrationRequestProgressDTO request : requests) {
+                    if (Objects.equals(request.getRegistrationStatus(), "APPROVED")) {
+                        return true;
+                    }
+                }
+            }
+            // Returns false if array was empty or "APPROVED" wasn't found
+            return false;
+
+        } catch (Exception e){
+            throw new CsdeException("Failed to check registration status for " + shortHash +
+                    " with exception " + e);
+        }
+    }
+
+    /**
+     * Polls for registration of a vnode
+     * @param shortHash short hash of the node which is being poled for
+     * @param duration the number of milliseconds before the pollign times out
+     * @param cooldown the number of milliseconds in between poll attempts.
+     */
+    private void pollForRegistration(String shortHash, int duration, int cooldown) throws CsdeException {
+
+        int timer = 0;
+        while (timer < duration ) {
```

Comment:
please review all files and stay consistent with while/for/if/else spacing

Reply:
Will do now for the new code, then the rest as part of the refactoring work.